R/validate-policy.R

```diff
@@ -1,3 +1,45 @@
+#' Validate Policy File
+#' 
+#' Flattens and validates policy file
+#'  @param policy YAML policy file
+#'  @param gs Gradescope data
+#'  @importFrom purrr map discard
+#'  @export
+validate_policy <- function(policy, gs){
+  policy <- flatten_policy(policy)
+  # drop categories with unavailable assignments
+  categories <- map(policy$categories, "category") |> unlist()
+  assignments <- c(get_assignments(gs), categories)
+  policy$categories <- map(policy$categories, function(cat){
+    cat$assignments <- cat$assignments[cat$assignments %in% assignments]
```

Comment:
perhaps add a comment in here because it's a bit obtuse what's going on here. Is it: for every category in the policy file, remove from the assignment list any assignments that aren't present in the gs data frame. If there are no assignments left in a category, remove it?

Reply:
Just added the comments. Some categories have only nested categories in their assignments, so they shouldn't be dropped